In [1]:
import pandas as pd
import json
import numpy as np

In [2]:
# Load business data
business = open("C:/Users/varal/Downloads/archive/yelp_academic_dataset_business.json",encoding='utf-8')
business_df = pd.DataFrame([json.loads(x) for x in business.readlines()])
business.close()


In [3]:
business_df.head()

business_id                      name  \
0  f9NumwFMBDn751xgFiRbNA  The Range At Lake Norman   
1  Yzvjg0SayhoZgCljUJRF9Q         Carlos Santo, NMD   
2  XNoUzKckATkOD1hP6vghZg                   Felinus   
3  6OAZjbxqM5ol29BuHsil3w      Nevada House of Hose   
4  51M2Kk903DFYI6gnB5I6SQ   USE MY GUY SERVICES LLC   

                     address             city state postal_code   latitude  \
0            10913 Bailey Rd        Cornelius    NC       28031  35.462724   
1  8880 E Via Linda, Ste 107       Scottsdale    AZ       85258  33.569404   
2      3554 Rue Notre-Dame O         Montreal    QC     H4C 1P4  45.479984   
3             1015 Sharp Cir  North Las Vegas    NV       89030  36.219728   
4         4827 E Downing Cir             Mesa    AZ       85205  33.428065   

    longitude  stars  review_count  is_open  \
0  -80.852612    3.5            36        1   
1 -111.890264    5.0             4        1   
2  -73.580070    5.0             5        1   
3 -115.127725    2.5             3        0   
4 -111.726648    4.5            26        1   

                                          attributes  \
0  {'BusinessAcceptsCreditCards': 'True', 'BikePa...   
1  {'GoodForKids': 'True', 'ByAppointmentOnly': '...   
2                                               None   
3  {'BusinessAcceptsCreditCards': 'True', 'ByAppo...   
4  {'BusinessAcceptsCreditCards': 'True', 'ByAppo...   

                                          categories  \
0  Active Life, Gun/Rifle Ranges, Guns & Ammo, Sh...   
1  Health & Medical, Fitness & Instruction, Yoga,...   
2                   Pets, Pet Services, Pet Groomers   
3  Hardware Stores, Home Services, Building Suppl...   
4  Home Services, Plumbing, Electricians, Handyma...   

                                               hours  
0  {'Monday': '10:0-18:0', 'Tuesday': '11:0-20:0'...  
1                                               None  
2                                               None  
3  {'Monday': '7:0-16:0', 'Tuesday': '7:0-16:0', ...  
4  {'Monday': '0:0-0:0', 'Tuesday': '9:0-16:0', '...

In [4]:
# Extracting Vegas infromation
business_df = business_df[business_df['city']== 'Las Vegas']

In [5]:
# There are 31,631 business in Las Vegas
# Extracing only restaurants information
categories = pd.Series(business_df['categories'])
business_df['flag'] = categories.str.contains('Restaurants', na=False)
business_df = business_df[business_df['flag'] == True]
business_df.shape

(6931, 15)

In [6]:
# Finally the dataset has information of 6,931 Restaurants from Las Vegas

In [7]:
# Load reviews  data
reviews = open("C:/Users/varal/Downloads/archive/yelp_academic_dataset_review.json",encoding='utf-8')
reviews_df = pd.DataFrame([json.loads(x) for x in reviews.readlines()])
reviews.close()

In [8]:
reviews_df.head()

review_id                 user_id             business_id  \
0  xQY8N_XvtGbearJ5X4QryQ  OwjRMXRC0KyPrIlcjaXeFQ  -MhfebM0QIsKt87iDN-FNw   
1  UmFMZ8PyXZTY2QcwzsfQYA  nIJD_7ZXHq-FX8byPMOkMQ  lbrU8StCq3yDfr-QMnGrmQ   
2  LG2ZaYiOgpr2DK_90pYjNw  V34qejxNsCbcgD8C0HVk-Q  HQl28KMwrEKHqhFrrDqVNQ   
3  i6g_oA9Yf9Y31qt0wibXpw  ofKDkJKXSKZXu5xJNGiiBQ  5JxlZaqCnk1MnbgRirs40Q   
4  6TdNDKywdbjoTkizeMce8A  UgMW8bLE0QMJDCkQ1Ax5Mg  IS4cv902ykd8wj1TR0N3-A   

   stars  useful  funny  cool  \
0    2.0       5      0     0   
1    1.0       1      1     0   
2    5.0       1      0     0   
3    1.0       0      0     0   
4    4.0       0      0     0   

                                                text                 date  
0  As someone who has worked with many museums, I...  2015-04-15 05:21:16  
1  I am actually horrified this place is still in...  2013-12-07 03:16:52  
2  I love Deagan's. I do. I really do. The atmosp...  2015-12-05 03:18:11  
3  Dismal, lukewarm, defrosted-tasting "TexMex" g...  2011-05-27 05:30:52  
4  Oh happy day, finally have a Canes near my cas...  2017-01-14 21:56:57

In [9]:
# extract business_id information
business = business_df['business_id']
# Merge reviews and business information
data = pd.merge(business_df, reviews_df, on='business_id', how='left')

In [10]:
# counting the number of reviews, each user has written and appending that value to the dataframe
data['user-reviewCount']= data['user_id'].map(data['user_id'].value_counts())

In [11]:
# Collecting the details of restaurants that has atleast 20 User-Review count
data = data[data['user-reviewCount']>=50]
data = data[data['user-reviewCount'] <= 500]

In [12]:
data.shape

(167633, 24)

In [13]:
len(set(data['business_id']))

6323

In [14]:
len(set(data['user_id']))

1808

In [15]:
data['user-reviewCount'].mean()

128.68114869983833

In [16]:
data['user-reviewCount'].max()

492

In [17]:
data['user-reviewCount'].min()

50

In [18]:
# counting the number of reviews, each user has written and appending that value to the dataframe
data['business-reviewCount']= data['business_id'].map(data['business_id'].value_counts())

In [19]:
data['business-reviewCount'].mean()

92.37711548442132

In [20]:
# Removing the business that has less than 70 reviews
x = data[data['business-reviewCount'] >= 70]
x.shape

(82826, 25)

##### There are total 662 business with 1807 users. There are total of 82826 reviews for 662 restauarnts with an average of  126 reviews for each restaurant

In [21]:
x.columns
data = x
data.columns

Index(['business_id', 'name', 'address', 'city', 'state', 'postal_code',
       'latitude', 'longitude', 'stars_x', 'review_count', 'is_open',
       'attributes', 'categories', 'hours', 'flag', 'review_id', 'user_id',
       'stars_y', 'useful', 'funny', 'cool', 'text', 'date',
       'user-reviewCount', 'business-reviewCount'],
      dtype='object')

In [23]:
y = x.sample(n=40000)
data = y

In [28]:
data.to_csv('AIT722-data.csv',index = False)

In [27]:
data.shape

(40000, 25)

In [24]:
len(set(y['user_id']))

1807

In [25]:
len(set(y['business_id']))

662

In [26]:
data['user-reviewCount'].mean()

126.49497742254847